src/components/app/MenuButtons/MetaInfo.js

```diff
@@ -21,6 +29,54 @@ type Props = {
 export class MenuButtonsMetaInfo extends React.PureComponent<Props> {
   render() {
     const meta = this.props.profile.meta;
+    const overheads: ?(ProfilerOverhead[]) = this.props.profile
+      .profilerOverhead;
+    const calculatedStats: Map<string, ProfilerStats> = new Map();
+    // These 3 values only have single values, so we are not using ProfilerStats class for them.
+    let overheadDurations = 0;
+    let overheadPercentage = 0;
+    let profiledDuration = 0;
+    let totalSamplingCount = 0;
+
+    // Older profiles(Before FF 70) don't have any overhead info. Don't show anything if
+    // that's the case.
+    if (overheads) {
+      // Overhead keys that have min/max/mean values to loop.
+      const statKeys = [
+        'Overhead',
+        'Cleaning',
+        'Counter',
+        'Interval',
+        'Lockings',
+      ];
+
+      for (const overhead of overheads) {
+        const { statistics } = overhead;
+        const { samplingCount } = statistics;
+
+        // Calculation the single values without any loop, it's not worth it.
+        overheadDurations += statistics.overheadDurations * samplingCount;
+        overheadPercentage += statistics.overheadPercentage * samplingCount;
+        profiledDuration += statistics.profiledDuration * samplingCount;
+        totalSamplingCount += samplingCount;
+
+        // Looping through the overhead values that have min/max/mean values
+        // and calculating them.
+        for (const stat of statKeys) {
+          const max = statistics['max' + stat];
+          const mean = statistics['mean' + stat];
+          const min = statistics['min' + stat];
+
+          let currentStat = calculatedStats.get(stat);
+          if (currentStat === undefined) {
+            currentStat = new ProfilerStats();
+            calculatedStats.set(stat, currentStat);
+          }
+
+          currentStat.count(min, max, mean, samplingCount);
+        }
```

Comment:
It's really not clear to me what we're trying to do here. I think writing a concrete example would help, as a comment to the class `ProfilerStats`.
Especially it's not clear to me why we override min/max/mean at each call.

Reply:
Also, have you validated by our primary user for this feature, aka Gerald, that this is the best way to present this to make it useful for him? For example is it useful to have 1 value instead of 1 per process?

Reply:
@squelart pinging to check with you. I believe weighted mean values were you wanted. Could you take a look and confirm that if this is useful for you?

Reply:
Yes, thank you, that's what I expected.

Though your example above `class ProfilerStats` has the wrong result: (90*10 + 10*100) / (10+100) = 17.3, not 19. 😉

Reply:
@squelart in the example, weights are 90 and 10(samplingCount). We should divide to the sum weight. So the calculation should be ` (90*10 + 10*100) / (10+90) = 19` 🙂